src/PowerShellEditorServices/Session/PowerShellContext.cs

```diff
@@ -511,13 +511,43 @@ public Task<RunspaceHandle> GetRunspaceHandle(CancellationToken cancellationToke
 
                         if (this.powerShell.HadErrors)
                         {
-                            string errorMessage = "Execution completed with errors:\r\n\r\n";
+                            // Get the command/params that we were trying execute as a string in order to log it
+                            string commandText = "";
+                            foreach (var cmd in psCommand.Commands)
+                            {
+                                commandText += cmd.CommandText;
+                                foreach (var param in cmd.Parameters)
+                                {
+                                    commandText += $" -{param.Name} {param.Value}";
+                                }
+                                commandText += ";";
+                            }
+
+                            var strBld = new StringBuilder(1024);
+                            strBld.Append($"Execution of command '{commandText}' completed with errors:\r\n\r\n");
 
+                            int i = 1;
                             foreach (var error in this.powerShell.Streams.Error)
                             {
-                                errorMessage += error.ToString() + "\r\n";
+                                if (i > 1) strBld.Append("\r\n\r\n");
+                                strBld.Append($"Error #{i++}:\r\n");
+                                strBld.Append(error.ToString() + "\r\n");
+                                strBld.Append("ScriptStackTrace:\r\n");
+                                strBld.Append(error.ScriptStackTrace + "\r\n");
+                                strBld.Append($"Exception {error.Exception.ToString()}");
+                                Exception innerEx = error.Exception?.InnerException;
+                                while (innerEx != null)
+                                {
+                                    strBld.Append($"InnerException {innerEx.ToString()}");
+                                    innerEx = innerEx.InnerException;
+                                }
                             }
 
+                            // We've reported these errors, clear them so they don't keep showing up.
+                            this.powerShell.Streams.Error.Clear();
```

Comment:
Weird, I could have sworn I added code for this in the past.

Reply:
And there very well could be a better place to put this call.  I contemplated clearing the error collection at the beginning of the method, before it runs the PSCommand.  But I stuck with only clearing after reporting error message.  BTW they weren't getting cleared.  I kept seeing the same two import failures ($pseditor null) for every subsequent execution - profile loading, prompt eval, etc.

Reply:
Nah, this is the right place to do it.  I just couldn't believe I hadn't actually added the code.  Probably was in a branch somewhere that never got merged :/